source/plugins/ruby/extension.rb

```diff
@@ -21,57 +21,101 @@ def get_output_stream_id(datatypeId)
       if @cache.has_key?(datatypeId)
         return @cache[datatypeId]
       else
-        @cache = get_config()
+        @cache = get_stream_mapping()
         return @cache[datatypeId]
       end
     }
   end
 
+  def get_extension_settings()
+    extensionSettings = Hash.new
+    begin
+      extensionConfigurations = get_extension_configs()
+      if !extensionConfigurations.nil? && !extensionConfigurations.empty?
+        extensionConfigurations.each do |extensionConfig|

```

Comment:
can there be multiple extensionConfigs?

Reply:
> can there be multiple extensionConfigs?
No, we only support single container insights extension and single extension config.